lib/process-services-cloud/src/lib/task/task-filters/components/task-assignment-filter/task-assignment-filter.component.spec.ts

```diff
@@ -54,34 +70,135 @@ describe('EditTaskFilterCloudComponent', () => {
+
+        const assignmentTypeOptions: DebugElement[] = fixture.debugElement.queryAll(By.css('mat-option'));
+
+        expect(assignmentTypeOptions.length).toEqual(5);
+        expect(assignmentTypeOptions[0].nativeElement.innerText).toEqual('ADF_CLOUD_TASK_ASSIGNMENT_FILTER.NONE');
+        expect(assignmentTypeOptions[1].nativeElement.innerText).toEqual('ADF_CLOUD_TASK_ASSIGNMENT_FILTER.UNASSIGNED');
+        expect(assignmentTypeOptions[2].nativeElement.innerText).toEqual('ADF_CLOUD_TASK_ASSIGNMENT_FILTER.ASSIGNED_TO');
+        expect(assignmentTypeOptions[3].nativeElement.innerText).toEqual('ADF_CLOUD_TASK_ASSIGNMENT_FILTER.CURRENT_USER');
+        expect(assignmentTypeOptions[4].nativeElement.innerText).toEqual('ADF_CLOUD_TASK_ASSIGNMENT_FILTER.CANDIDATE_GROUPS');
     });
 
-    it('should show the candidate groups', () => {
-        component.assignmentType = AssignmentType.CANDIDATE_GROUPS;
-        fixture.detectChanges();
-        const candidateGroups = fixture.debugElement.nativeElement.querySelector('.adf-group-cloud-filter');
+    it('should emit the current user info when assignment is the current user', () => {
+        spyOn(identityUserService, 'getCurrentUserInfo').and.returnValue(mockIdentityUsers[0]);
+        spyOn(component.assignedUsersChange, 'emit');
+
+        selectAssignmentType(AssignmentType.CURRENT_USER);
+
+        expect(component.assignedUsersChange.emit).toHaveBeenCalledWith([mockIdentityUsers[0]]);
+    });
+
+    it('should show the CANDIDATE_GROUPS input', () => {
+        selectAssignmentType(AssignmentType.CANDIDATE_GROUPS);
+
+        const candidateGroups = fixture.debugElement.query(By.css('[data-automation-id="adf-group-cloud-candidate-groups-filter"]'));
+        expect(component.candidateGroups.length).toEqual(0);
         expect(candidateGroups).toBeTruthy();
     });
 
+    it('should show the ASSIGNED_TO input', () => {
+        selectAssignmentType(AssignmentType.ASSIGNED_TO);
+
+        const candidateGroups = fixture.debugElement.query(By.css('[data-automation-id="adf-group-cloud-assigned-to-filter"]'));
+        expect(component.assignedUsers.length).toEqual(0);
+        expect(candidateGroups).toBeTruthy();
+    });
+
+    describe('status input change', () => {
+        it('should CREATED status set assignment type to UNASSIGNED', () => {
+            const createdStatusChange = new SimpleChange(null, 'CREATED', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.UNASSIGNED);
+        });
+
+        it('should ASSIGNED status set assignment type to ASSIGNED_TO', () => {
+            const createdStatusChange = new SimpleChange(null, 'ASSIGNED', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.ASSIGNED_TO);
+        });
+
+        it('should empty status set assignment type to NONE', () => {
```

Comment:
what's empty status? Something like ALL?

Reply:
yes, it's equal ALL in this case. changed to ALL (+ used enum instead of string)